.github/workflows/workflow.yml

```diff
@@ -142,363 +129,265 @@ jobs:
-    - name: Kind cluster cleanup
-      env:
-        DOCKER_HOST: ssh://linkerd-docker
-      run: |
-        TAG="$(CI_FORCE_CLEAN=1 bin/root-tag)"
-        export KIND_CLUSTER=github-$TAG-${{ matrix.integration_test }}
-        bin/kind delete cluster --name=$KIND_CLUSTER
+          . bin/_docker.sh
+          echo ::set-env name=DOCKER_REGISTRY::$DOCKER_REGISTRY
+      - name: Setup SSH config for Packet
+        if: github.event_name != 'pull_request' || !github.event.pull_request.head.repo.fork
+        run: |
+          mkdir -p ~/.ssh/
+          touch ~/.ssh/id && chmod 600 ~/.ssh/id
+          echo "${{ secrets.DOCKER_SSH_CONFIG }}"  > ~/.ssh/config
+          echo "${{ secrets.DOCKER_PRIVATE_KEY }}" > ~/.ssh/id
+          echo "${{ secrets.DOCKER_KNOWN_HOSTS }}" > ~/.ssh/known_hosts
+      - name: Download image archives (Forked repositories)
+        if: github.event_name == 'pull_request' && github.event.pull_request.head.repo.fork
+        uses: actions/download-artifact@v1
+        with:
+          name: image-archives
+      - name: Load cli-bin image into local docker images
+        if: github.event_name != 'pull_request' || !github.event.pull_request.head.repo.fork
+        run: |
+          # `docker load` only accepts input from STDIN, so pipe the image
+          # archive into the command.
+          #
+          # In order to pipe the image archive, set `DOCKER_HOST` for a single
+          # command and `docker save` the CLI image from the Packet host.
+          DOCKER_HOST=ssh://linkerd-docker docker save "$DOCKER_REGISTRY/cli-bin:$TAG" | docker load
+      - name: Load cli-bin image into local docker images (Forked repositories)
+        if: github.event_name == 'pull_request' && github.event.pull_request.head.repo.fork
+        run: docker load < image-archives/cli-bin.tar
+      - name: Install CLI
+        run: |
+          # Copy the CLI out of the local cli-bin container.
+          container_id=$(docker create "$DOCKER_REGISTRY/cli-bin:$TAG")
+          docker cp $container_id:/out/linkerd-linux $HOME/.linkerd
+
+          # Validate the CLI version matches the current build tag.
+          [[ "$TAG" == "$($HOME/.linkerd version --short --client)" ]]
+      - name: Setup default KinD cluster
+        if: matrix.integration_test != 'custom_domain'
+        uses: engineerd/setup-kind@v0.3.0
+        with:
+          version: "v0.6.1"
+      - name: Setup custom_domain KinD cluster
+        if: matrix.integration_test == 'custom_domain'
+        uses: engineerd/setup-kind@v0.3.0
+        with:
+          config: test/testdata/custom_cluster_domain_config.yaml
+          version: "v0.6.1"
+      - name: Load image archives into the local KinD cluster
+        if: github.event_name != 'pull_request' || !github.event.pull_request.head.repo.fork
+        env:
+          PROXY_INIT_IMAGE_NAME: gcr.io/linkerd-io/proxy-init:v1.3.1
+          PROMETHEUS_IMAGE_NAME: prom/prometheus:v2.11.1
```

Comment:
We've been on prom v2.15.2 for a while, not sure how has this been working without updating this tag...

Reply:
Hm yea I'm not sure. I can update it, but maybe as a separate PR? I'd like to keep a version bump out of this if possible.

Reply:
This image name is for seeding the kind cluster docker caches prior to Linkerd installation. It makes CI (and Linkerd's bootup) faster. It should be kept in-sync with Linkerd, but worst case it just slows things down.